dspinner/pin.go

```diff
@@ -164,7 +164,13 @@ func New(ctx context.Context, dstore ds.Datastore, dserv ipld.DAGService) (ipfsp
 }
 
 // Pin the given node, optionally recursive
-func (p *pinner) Pin(ctx context.Context, node ipld.Node, recurse bool) error {
+func (p *pinner) Pin(ctx context.Context, node ipld.Node, recurse bool) (perr error) {
+	defer func() {
+		if perr == nil {
+			perr = p.Flush(ctx)
+		}
+	}()
```

Comment:
This is likely ok, but sometimes we return early (e.g. if the data is already pinned) in which case we are flushing unnecessarily because we made no changes.

Reply:
Made it more precise. Added comment. In https://github.com/ipfs/go-ipfs-pinner/pull/13/commits/a24f5872bea10d592361febcd6bb3bf0ea5743e0.